src/python/WMCore/Database/DBCore.py

```diff
@@ -156,15 +163,28 @@ def processData(self, sqlstmt, binds={}, conn=None,
                 #Run single SQL statement for a list of binds - use execute_many()
                 if not transaction:
                     trans = connection.begin()
-                while(len(binds) > self.maxBindsPerQuery):
-                    result.extend(self.processData(sqlstmt, binds[:self.maxBindsPerQuery],
-                                                   conn=connection, transaction=True,
-                                                   returnCursor=returnCursor))
-                    binds = binds[self.maxBindsPerQuery:]
 
-                for i in sqlstmt:
-                    result.extend(self.executemanybinds(i, binds, connection=connection,
-                                                        returnCursor=returnCursor))
+                s = sqlstmt[0]
+
+                if s.lower().endswith('select', 0, 6) and returnCursor:
+                    iterableFlag = True
+                else:
+                    iterableFlag = False
+
+                if iterableFlag:
+                    # for the select statement with returnCursor true, don't worry about sending to many bindings
+                    # it sends one
+                   result = self.executemanybinds(s, binds, connection=connection,
+                                                        returnCursor=returnCursor)
+                else:
+                    while (len(binds) > self.maxBindsPerQuery):
+                        result.extend(self.processData(sqlstmt, binds[:self.maxBindsPerQuery],
```

Comment:
I guess I'm uncomfortable with requiring `processData` to know low-level implementation details of `executemanybinds`.  Can you factor out the logic into a shared method?

Reply:
The problem is executemanybinds returns different type depending on the condition. (Iterable from generator and list). I could change to make return iterable not list but since this function is used everywhere (outside WMCore (DBS) as well), I have to coordinate no code expect list.
Also I am not sure I how I can deal with recursive call with the function returning iterable. (We have to yield again be second yield will return the generator object not the original object yield from generator.

FYI (This recursive call is originally added for the limitation of number of binding variable Oracle accepts - I think it is 500(?))

We can completely refactor code, 
1. This only returns iterable of cursors - too many changes
2. Separate select and insert operation give different executemany function for that.